go-controller/pkg/ovn/service.go

```diff
@@ -275,34 +264,39 @@ func (ovn *Controller) createService(service *kapi.Service) error {
 					}
 				} else {
 					aclUUID, err := ovn.createLoadBalancerRejectACL(loadBalancer, service.Spec.ClusterIP,
-						svcPort.Port, protocol)
+						svcPort.Port, svcPort.Protocol)
 					if err != nil {
 						return fmt.Errorf("failed to create service ACL: %v", err)
 					} else {
 						klog.V(5).Infof("Service Reject ACL created for cluster IP: %s", aclUUID)
 					}
 				}
 				for _, extIP := range service.Spec.ExternalIPs {
-					exLoadBalancer := ovn.getDefaultGatewayLoadBalancer(svcPort.Protocol)
-					if exLoadBalancer == "" {
-						klog.Warningf("No default gateway found for protocol %s\n\tNote: 'nodeport'"+
-							"flag needs to be enabled for default gateway", svcPort.Protocol)
-						continue
+					gateways, _, err := ovn.getOvnGateways()
+					if err != nil {
+						return err
 					}
-					vip := util.JoinHostPortInt32(extIP, svcPort.Port)
-					// Skip creating LB if endpoints watcher already did it
-					if _, hasEps := ovn.getServiceLBInfo(loadBalancer, vip); hasEps {
-						klog.V(5).Infof("Load Balancer already configured for %s, %s", loadBalancer, vip)
-					} else if ep != nil {
-						if err := ovn.AddEndpoints(ep); err != nil {
-							return err
-						}
-					} else {
-						aclUUID, err := ovn.createLoadBalancerRejectACL(exLoadBalancer, extIP, svcPort.Port, protocol)
+					for _, gateway := range gateways {
+						loadBalancer, err := ovn.getGatewayLoadBalancer(gateway, svcPort.Protocol)
 						if err != nil {
-							return fmt.Errorf("failed to create service ACL for external IP")
+							klog.Errorf("physical gateway %s does not have load_balancer "+
+								"(%v)", gateway, err)
+							continue
+						}
+						if loadBalancer == "" {
+							continue
+						}
+						vip := util.JoinHostPortInt32(extIP, svcPort.Port)
+						// Skip creating LB if endpoints watcher already did it
+						if _, hasEps := ovn.getServiceLBInfo(loadBalancer, vip); hasEps {
```

Comment:
dont you need the check that you added in your "small nit changes" pr to check eps like you did for cluster service ip:
```
if _, hasEps := ovn.getServiceLBInfo(loadBalancer, vip); hasEps {
					klog.V(5).Infof("Load Balancer already configured for %s, %s", loadBalancer, vip)
				} else if ep != nil {

```

Reply:
No, not here. Because if it will already been done for the done on L243 above for the `ClusterIP`

Reply:
ok i see we return there